plugins/uefi/fu-plugin-uefi.c

```diff
@@ -758,13 +758,20 @@ fu_plugin_coldplug (FuPlugin *plugin, GError **error)
 		return FALSE;
 
 	/* make sure that efivarfs is rw */
-	if (!fu_plugin_uefi_ensure_efivarfs_rw (&error_efivarfs))
-		g_warning ("%s", error_efivarfs->message);
+	if (!fu_plugin_uefi_ensure_efivarfs_rw (&error_efivarfs)) {
+		fu_plugin_set_mode (plugin, FU_PLUGIN_MODE_RT_DEGRADED);
+		/* FIXME: should this be fu_plugin_set_*error*_text */
+		fu_plugin_set_mode_text (plugin, error_efivarfs->message);
+	}
 
 	if (data->esp == NULL) {
 		data->esp = fu_common_get_esp_default (&error_udisks2);
-		if (data->esp == NULL)
-			g_warning ("cannot find default ESP: %s", error_udisks2->message);
+		if (data->esp == NULL) {
+			fu_plugin_set_mode (plugin, FU_PLUGIN_MODE_RT_DEGRADED);
+			/* FIXME: should this overwrite the EFIVAR error?
+			 * should this be fu_plugin_*add*_mode_text instead? */
+			fu_plugin_set_mode_text (plugin, error_udisks2->message);
+		}
```

Comment:
I guess we need to decide if there will still be dummy devices or not to decide if two runtime degraded reasons are valid.

Reply:
> two runtime degraded reasons are valid

I think they very much are. Thanks.